spec/integration/action_spec.rb

```diff
@@ -428,4 +428,51 @@ def b
       end
     end
   end
+
+  describe 'optional values' do
+    let(:complex_action) do
+      Class.new do
+        include Verbalize::Action
+        input optional: [:no_default, default_const: 3, default_method: ->() { Date.today }]
+
+        def call
+          {
+            no_default:     no_default,
+            default_const:  default_const,
+            default_method: default_method
+          }
+        end
+      end
+    end
+
+    it 'wraps value consts in methods' do
+      expect(complex_action.defaults[:default_const]).to respond_to(:call)
+    end
+
+    it 'renders default values if no value is passed in' do
+      expect(complex_action.call!).to eq(
+        no_default:     nil,
+        default_const:  3,
+        default_method: Date.today
+      )
+    end
```

Comment:
arguably unnecessary since we put the defaults in as default *arguments* on the eval-built `class#call`, and we know for a fact how ruby handles default values, but sure.